### DISCUSSION TITLE
Welcome to stm32-usart-uart-dma-rx-tx Discussions!

### DISCUSSION THREAD

**Discussion by MaJerle** [MaJerle/STM32_USART_DMA_RX#12] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by provitek** [comment]
Hi :)
I tried to port the code usart_rx_idle_line_irq_ringbuff_tx_H7 to h723 but it does not work for me.
I did a fresh cubemx project for h723 nucleo board, LL for UART, DMA etc, added user code from h743 project and effect is: no DMA interrupts are triggered.
This is my simple mistake when porting the code, or these controllers differs and the code needs to be changed - e.g. other flags, LL libraries?
Regards 
Chris

**Reply by MaJerle** [reply]
Please compare linker script (.LD) of my project and then one from your project. You will notice that whatever RAM is used by application, it is the one from D1 domain. By default, DTCM RAM is used (fastest one) and DMA has no access to this RAM, hence not working and no interrupt.

Read this FAQ: https://community.st.com/s/article/FAQ-DMA-is-not-working-on-STM32H7-devices
It will give you good overview what I'm talking about.

In my sample demo, to avoid problems, I simply remapped RAM address to reflect D1 memory and problem was solved.

**Reply by provitek** [reply]
Thank you for tips. 
I have changed all RAM to RAM_EXEC but it still does not work :/
I'm quite new to STM32 and forgive me maybe trivial questions
Could you see attached file and check if it is correct?
[STM32H723ZGTX_RAM.ld.txt](https://github.com/MaJerle/stm32-usart-uart-dma-rx-tx/files/6204895/STM32H723ZGTX_RAM.ld.txt)

**Reply by MaJerle** [reply]
But why did you delete flash? See my linker script and compare yours. Line by line.

**Reply by provitek** [reply]
I did not :) 
I don't knew I have to edit it. 
Very BIG THANK YOU!!!! All works!
